### DISCUSSION TITLE
Next Run field in Agent Jobs tab

### DISCUSSION THREAD

**Discussion by tdinh2011** [trimble-oss/dba-dash#599] [Ideas]
Hello,

While looking through the Agent Jobs tab, I did not see info about the next scheduled run date/time. It would helpful to include this field. 

Thanks,
Tuan

**Comment by DavidWiseman** [comment]
Thank you for the suggestion.  The next run date could be obtained from [msdb.dbo.sysjobsctivity](https://learn.microsoft.com/en-us/sql/relational-databases/system-tables/dbo-sysjobactivity-transact-sql?view=sql-server-ver16).  If we use this to add the next scheduled run date to the jobs collection, this only runs every 1hr by default.  It could be confusing for more frequently executed jobs to show a date in the past.  Another option would be to calculate this from the schedules. This would be more complex and might be error prone if the calculations don't exactly match.
It could be worth having a totally separate collection for sysjobactivity that runs more frequently.  This could also be used to identify jobs that are currently running.

**Comment by tdinh2011** [comment]
Thanks for quick response, David. The idea came up as I was troubleshooting a failed job in DBA Dash. I wanted to know when is the next scheduled date/time for the job within the tool so if it was just a one-off failure, I can let it try again and it would be helpful to have the schedule all in one place.
